bug(COMPONENT): ng update failes without spec files

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

_No response_

### Description

ng update @angular/cdk from 15.x to 16 fails if there are no spec.ts or d.ts files in the app folder. This is the case for projects that are created with --skip-tests. 

Workarround is: create an empty d.ts file

### Reproduction

Steps to reproduce:
1. create a project using --skip-tests
2. make sure that there are no d.ts or spec.ts files
3. run ng update including material or cdk 


### Expected Behavior

migrations run

### Actual Behavior

✖ Migration failed: error TS18003: No inputs were found in config file 'tsconfig.json'. Specified 'include' paths were '["src/**/*.spec.ts","src/**/*.d.ts"]' and 'exclude' paths were '["./out-tsc/spec"]'.

### Environment

- Angular: 15.x
- CDK/Material: 15.x
- Browser(s): none
- Operating System (e.g. Windows, macOS, Ubuntu): macOS

--- a/src/cdk/schematics/update-tool/utils/parse-tsconfig.ts
+++ b/src/cdk/schematics/update-tool/utils/parse-tsconfig.ts
@@ -12,6 +12,9 @@ import {FileSystemHost} from './virtual-host';
 import {dirname} from 'path';
 import {formatDiagnostics} from './diagnostics';
 
+/** Code of the error raised by TypeScript when a tsconfig doesn't match any files. */
+const NO_INPUTS_ERROR_CODE = 18003;
+
 /** Class capturing a tsconfig parse error. */
 export class TsconfigParseError extends Error {}
 
@@ -45,8 +48,12 @@ export function parseTsconfigFile(
     {},
   );
 
-  if (parsed.errors.length) {
-    throw new TsconfigParseError(formatDiagnostics(parsed.errors, fileSystem));
+  // Skip the "No inputs found..." error since we don't want to interrupt the migration if a
+  // tsconfig doesn't match a file. This will result in an empty `Program` which is still valid.
+  const errors = parsed.errors.filter(diag => diag.code !== NO_INPUTS_ERROR_CODE);
+
+  if (errors.length) {
+    throw new TsconfigParseError(formatDiagnostics(errors, fileSystem));
   }
 
   return parsed;

